### PR TITLE
Mount out

### PR DESCRIPTION
**Pull request by pizzamig**
To address issue #170 , this PR add the ability to:
* unmount a mountpoint from a running pot (if the mountpoint is not "in use")
* remove the mount entry from the pot configuration

### CODE REVIEW

File: share/pot/mount-out.sh [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# $1 pot
+# $2 mount point
+_is_mountpoint_used()
+{
+	local _pname _mnt_p _proot
+	_pname="$1"
+	_mnt_p="${2#/}"
+	_conf=$POT_FS_ROOT/jails/$_pname/conf/fscomp.conf
+	_proot=$POT_FS_ROOT/jails/$_pname/m
+	if grep -q " $_proot/$_mnt_p$" "$_conf" ||
```
Comment (grembo) [inline share/pot/mount-out.sh:23]: I assume the space is intentional " ", maybe a comment would be helpful?

File: share/pot/mount-out.sh [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+		grep -q " $_proot/$_mnt_p " "$_conf" ; then
+		# mount point already used
+		return 0 # true
+	fi
+	if grep -q "$_proot/$_mnt_p " "$_conf" ; then
```
Comment (grembo) [inline share/pot/mount-out.sh:28]: Maybe explicitly grepping for begin of line (like "^blabla") would be useful here? Also, what about mountpoints containing spaces?

File: share/pot/mount-out.sh (hunk 1 above)
Comment (pizzamig) [reply share/pot/mount-out.sh:26]: Those spaces are there to enforce an exact match.
A pattern like `/mnt/test` would match both `/mnt/test` and `/mnt/test2`.
Those spaces are there to prevent `/mnt/test2` to be a match

File: share/pot/mount-out.sh (hunk 2 above)
Comment (pizzamig) [reply share/pot/mount-out.sh:31]: this string is not at the begin of the line, so ^ cannot be used.
mountpoints with space shouldn't be an issue, the double quote should preserve everything correctly.

File: share/pot/mount-out.sh (hunk 1 above)
Comment (grembo) [reply share/pot/mount-out.sh:26]: I understood that, but only as I was reviewing the change, hence asking for the comment.

File: share/pot/mount-out.sh (hunk 2 above)
Comment (grembo) [reply share/pot/mount-out.sh:31]: I though this grep is looking for exactly that ($_proot/$_mnt_p being used as a source, which I though was on the left hand side of fscomp.conf)? Or could there be cases where quotes exist left of the mount source?

My comment regarding spaces was about a situation where a mountpoint named "/opt/pot/jails/test/m/mnt with space" would also be matched by this grep for a mountpoint named  "/opt/pot/jails/test/m/mnt" (as the grep would be for "/opt/pot/jails/test/m/mnt "). Probably a corner case, but it still would be nice if such things could be avoided programmatically.

Comment (pizzamig) [reply share/pot/mount-out.sh:31]: `$_proot/$_mnt_p` is the second element in the file, it's the mountpoint inside the jail (the destination).
The same source could be mounted multiple times, while you can only have one mountpoint.

About your corner case, yes, you are right, it's pathological, that means valid and possible.
Do you have nay suggestion on how to "easily" tackle it?
The only solutions I have in mind are:
* ask the user if a grep returns multiple entries (not script-friendly, tho)
* match the entire string, from ^ to $. However, in this case, we would need to ask the user to input the source and the options (a bit annoying IMO)

I'm open to any suggestion :)

Comment (grembo) [reply share/pot/mount-out.sh:31]: > $_proot/$_mnt_p is the second element in the file, it's the mountpoint inside the jail (the destination).

I thought that this is what the first grep was about, seems like I'm confused.

> I'm open to any suggestion :)

My solution would be to simply not allow whitespace characters in mountpoints and have a regex check enforcing that.
